elasticdl/python/worker/task_data_service.py

```diff
@@ -123,14 +128,8 @@ def _gen(self):
                 )
                 if len(self._pending_tasks_with_counts) == 1:
                     self._current_task = task
-            with closing(
-                recordio.Scanner(
-                    task.shard_name, task.start, task.end - task.start
-                )
-            ) as reader:
-                while True:
-                    record = reader.record()
-                    if record:
-                        yield record
-                    else:
-                        break
+            for data in self._data_reader.read_records(task):
+                if data:
+                    yield data
```

Comment:
Can we make `read_records` into a generator? If no data is returned, raise `StopIteration`?

Reply:
Good idea. I'll improve this later.